service-deployment/src/grafana/dockerfile.template

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+FROM debian:jessie
+MAINTAINER Yitong Feng imfengyitong@gmail.com
+
+### GRAFANA_VERSION=latest = nightly build
+ENV \
+  GRAFANA_URL={{ clusterconfig['clusterinfo']['prometheusinfo']['grafana_addr'] }} \
+  PROM_URL={{ clusterconfig['clusterinfo']['prometheusinfo']['prometheus_addr'] }} \
+  GRAFANA_VERSION=4.6.2 \
+  GF_PLUGIN_DIR=/grafana-plugins \
+  GF_PATHS_LOGS=/var/log/grafana \
+  GF_PATHS_DATA=/var/lib/grafana \
+  UPGRADEALL=false
+
+
+COPY ./run.sh /run.sh
+COPY ./grafana_config.sh /grafana_config.sh
+COPY ./start_server.sh /start_server.sh
+COPY ./prom-datasource.json /
+COPY ./dashboards/*.json /dashboards/
+
+RUN \
+  apt-get update && \
+  apt-get -y --force-yes --no-install-recommends install libfontconfig curl ca-certificates git jq && \
+  curl https://s3-us-west-2.amazonaws.com/grafana-releases/release/grafana_${GRAFANA_VERSION}_amd64.deb > /tmp/grafana.deb && \
```

Comment:
>s3-us-west-2.amazonaws.com [](start = 15, length = 26)

I doubt this the official place to get grafana. if cannot get it through apt-get, I suggest to find it from grafana official site

Reply:
It's official site.